crates/lair_keystore_api/src/sodium_secretstream.rs

```diff
@@ -192,72 +241,123 @@ where
-        use sodoken::kx;
-
         // box these up into trait objects so we can easily refer to their types.
         let mut send: PrivRawSend = Box::new(send);
         let mut recv: PrivRawRecv = Box::new(recv);
 
         // generate an ephemeral cbox keypair
-        let eph_cbox_pub = sodoken::BufWriteSized::new_no_lock();
-        let eph_cbox_sec = sodoken::BufWriteSized::new_mem_locked()?;
-        cbox::keypair(eph_cbox_pub.clone(), eph_cbox_sec.clone()).await?;
+        let (eph_cbox_pub, mut eph_cbox_sec) =
+            tokio::task::spawn_blocking(move || -> LairResult<_> {
+                let mut eph_cbox_pub =
+                    [0; sodoken::crypto_box::XSALSA_PUBLICKEYBYTES];
+                let mut eph_cbox_sec = sodoken::SizedLockedArray::<
+                    { sodoken::crypto_box::XSALSA_SECRETKEYBYTES },
+                >::new()?;
+                sodoken::crypto_box::xsalsa_keypair(
+                    &mut eph_cbox_pub,
+                    &mut eph_cbox_sec.lock(),
+                )?;
+
+                Ok((eph_cbox_pub, eph_cbox_sec))
+            })
+            .await
+            .map_err(OneErr::new)??;
 
         // generate an ephemeral kx keypair
-        let eph_kx_pub = sodoken::BufWriteSized::new_no_lock();
-        let eph_kx_sec = sodoken::BufWriteSized::new_mem_locked()?;
-        kx::keypair(eph_kx_pub.clone(), eph_kx_sec.clone())?;
+        let (eph_kx_pub, mut eph_kx_sec) =
+            tokio::task::spawn_blocking(move || -> LairResult<_> {
+                let mut eph_kx_pub =
+                    [0; sodoken::crypto_box::XSALSA_PUBLICKEYBYTES];
+                let mut eph_kx_sec = sodoken::SizedLockedArray::<
+                    { sodoken::crypto_box::XSALSA_SECRETKEYBYTES },
+                >::new()?;
+                sodoken::crypto_box::xsalsa_keypair(
+                    &mut eph_kx_pub,
+                    &mut eph_kx_sec.lock(),
+                )?;
+
+                Ok((eph_kx_pub, eph_kx_sec))
+            })
+            .await
+            .map_err(OneErr::new)??;
 
         // sealed initiator message
         let mut message: [u8; 64] = [0; 64];
-        message[..32].copy_from_slice(&eph_cbox_pub.read_lock());
-        message[32..].copy_from_slice(&eph_kx_pub.read_lock());
-        let message = sodoken::BufReadSized::from(message);
-        let cipher =
-            <sodoken::BufWriteSized<{ 64 + cbox::SEALBYTES }>>::new_no_lock();
-        cbox::seal(cipher.clone(), message, srv_id_pub_key).await?;
-        let cipher = cipher.try_unwrap_sized().unwrap();
+        message[..32].copy_from_slice(&eph_cbox_pub);
+        message[32..].copy_from_slice(&eph_kx_pub);
+
+        let mut cipher =
+            tokio::task::spawn_blocking(move || -> LairResult<_> {
+                let mut cipher = sodoken::SizedLockedArray::<
+                    { 64 + sodoken::crypto_box::XSALSA_SEALBYTES },
+                >::new()?;
```

Comment:
```suggestion
                let mut cipher = [0; 64 + sodoken::crypto_box::XSALSA_SEALBYTES];
```

cipher is encrypted

Reply:
https://github.com/holochain/lair/pull/143/commits/b6f64ac499f6c25e33b6b125e4b3e59e90761dca